themes/dark/assets/js/templates/root-dirs.vue

```diff
@@ -0,0 +1,225 @@
+                            return accumlator;
+                        }, []);
+                    return;
+                }
+
+                // Update selected root dir path and select it
+                this.rootDirs.find(rd => rd.selected).path = path;
+                this.selectedRootDir = path;
+
+                this.saveRootDirs();
+            }, { initialDir: this.selectedRootDir });
+        },
+        remove() {
+            // Remove a root dir
+            const oldDirIndex = this.rootDirs.findIndex(rd => rd.selected);
+            const oldDirPath = this.selectedRootDir;
+
+            // What would the list be like without the root dir we're removing?
+            const filteredRootDirs = this.rootDirs.filter(rd => !rd.selected);
+
+            // Pick a new selection, or null
+            if (filteredRootDirs.length > 0) {
+                const newSelected = (oldDirIndex > 0) ? oldDirIndex - 1 : 0;
+                this.selectedRootDir = filteredRootDirs[newSelected].path;
+            } else {
+                this.selectedRootDir = null;
+            }
+
+            // If we're deleting the current default root dir, pick a new default, or null
+            if (this.defaultRootDir !== null && oldDirPath === this.defaultRootDir) {
+                this.defaultRootDir = this.selectedRootDir;
+            }
+
+            // Finally, remove the root dir from the list
+            this.rootDirs = filteredRootDirs;
+
+            this.saveRootDirs();
+        },
+        setDefault() {
+            if (this.selectedRootDir === this.defaultRootDir) return;
+            this.defaultRootDir = this.selectedRootDir;
+            this.saveRootDirs();
+        },
+        async saveRootDirs() {
+            let rootDirs = [];
+            if (this.defaultRootDir !== null && this.rootDirs.length) {
+                let defaultIndex;
+                const paths = this.rootDirs.map((rd, index) => {
+                    if (rd.default) defaultIndex = index;
+                    return rd.path
+                });
+                rootDirs = Array.from(defaultIndex.toString()).concat(paths);
+            }
+            await api.patch('config/main', { rootDirs });
+        }
+    },
+    watch: {
+        rootDirs: {
+            handler() {
+                this.$emit('update:root-dirs', this.rootDirs);
+                this.$nextTick(() => {
+                    // @FIXME: Temporarily trigger a regular event as well
+                    $('#rootDirs').trigger('change');
+                });
+            },
+            deep: true,
+            immediate: false
+        },
+        rootDirsValue() {
+            this.$emit('update:root-dirs-value', this.rootDirsValue, this.rootDirs);
+            this.$nextTick(() => {
+                // @FIXME: Temporarily trigger a regular event as well
+                $('#rootDirText').trigger('change');
+            });
+        }
+    }
+};
+</script>
+<style>
```

Comment:
Should we make sfc css scoped?

Reply:
If you don't want the outside CSS to be able to override the component then yes.